### PR TITLE
OGM-910 Index and unique constraint creation for MongoDB

### PR DESCRIPTION
**Pull request by gsmet**
https://hibernate.atlassian.net/browse/OGM-910

This is a followup of https://github.com/hibernate/hibernate-ogm/pull/689 with a different approach as discussed with Gunnar and Emmanuel this morning.

I hope it will have more success :).

### CODE REVIEW

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/UniqueConstraintTest.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+		session.delete( entityWithConstraints );
+		tx.commit();
+		session.close();
+	}
+
+	@Test
+	public void shouldThrowExceptionForDuplicatedNaturalId() throws Throwable {
+		thrown.expect( HibernateException.class );
+		thrown.expectMessage( "OGM001230" );
+		thrown.expectMessage( "UK_r4dom13cd9fcnuy5i2ku9mp07" );
```
Comment (DavideD) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/UniqueConstraintTest.java]: Out of curiosity, how are these messages generated?

Comment (DavideD) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/UniqueConstraintTest.java]: Never mind, found it

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/692

The only thing is missing is the documentation. Should I create a new issue for it, or are you going to send a followup PR?

**Comment by gsmet** [comment]
I will send a followup PR. Just wanted to have an agreement on the implementation before writing the docs.

**Pull request closed without merging**